templates/etc/init.d/elasticsearch.OpenBSD.erb

```diff
@@ -0,0 +1,26 @@
+#!/bin/sh
+#
+# $OpenBSD: elasticsearch.rc,v 1.2 2015/07/15 23:34:49 ajacoutot Exp $
+
+daemon="/usr/local/elasticsearch/bin/elasticsearch"
+daemon_flags="-d -Des.config=/etc/elasticsearch/<%= @name %>/elasticsearch.yml -p <%= @pid_dir %>/elasticsearch-<%= @name %>.pid"
```

Comment:
I only have the few parameters in the file for the daemon_flags, since this is what the "original" rc script from the package sets. Therefore that is all 
Further, there is no concept of /etc/{sysconfig|defaults}/progname or the like like on many (or most) Linux distributions. 
That's why I protected the augeas call in config.pp "${elasticsearch::params::defaults_location}/elasticsearch" to only get evaluated, when that defaults_location parameter is set, which it isn't for OpenBSD.
Also I ripped all the defaults_file related stuff out of service/openbsd.pp.
However, the $service_flags parameter of each instance, the user of the module can override the daemon_flags of each rc script.

i.e. 
service { 'elasticsearch_es01':
  flags => '-d -Des.config=/etc/elasticsearch/es01/elasticsearch.yml -p /var/run/elasticsearch/elasticsearch-es01.pid-Des.default.path.home=/tmp/blubb -Des.default.path.logs=/var/log/mylogs/'
}

that would allow the user to specify whatever they like it to be.
OpenBSD users are usually used to those service flags.

Alternatively, as I did, I configured it for the instances in Hiera yaml:
elasticsearch::instances:
  'es01':
    datadir: '/var/elasticsearch/es-data-es01/'
    logging_level: 'DEBUG'
    config:
      path:
        conf: '/etc/elasticsearch/es01/'
        logs: '/var/log/elasticsearch/'

Both ways should work fine.